langchain/src/document_loaders/web/notiondb.ts

```diff
@@ -0,0 +1,208 @@
+import { BaseDocumentLoader } from "../base.js";
+import { Document } from "../../document.js";
+
+const NOTION_BASE_URL = "https://api.notion.com/v1";
+
+export interface NotionDBLoaderParams {
+  integrationToken: string;
+  databaseId: string;
+}
+
+interface NotionPage {
+  id: string;
+  object: "page";
+  created_time: string;
+  last_edited_time: string;
+  archived: boolean;
+  parent: {
+    type: "database_id" | "page_id";
+    database_id?: string;
+    page_id?: string;
+  };
+  properties: {
+    // @see https://developers.notion.com/reference/retrieve-a-page
+    [propertyName: string]: any;
+  };
+}
+
+export class NotionDBLoader
+  extends BaseDocumentLoader
+  implements NotionDBLoaderParams
+{
+  public integrationToken: string;
+
+  public databaseId: string;
+
+  private headers: Record<string, string> = {};
+
+  constructor({ integrationToken, databaseId }: NotionDBLoaderParams) {
+    super();
+
+    if (!integrationToken) {
```

Comment:
Would be nice to try to load this from an environment variable like `process.env.NOTION_INTEGRATION_TOKEN`:

https://github.com/hwchase17/langchainjs/pull/1396/files#diff-e709c029fd4e48f9d3c541ae2399d64278cfe86491d525cd2e9d9f146281f3a8R33